### PR TITLE
vkd3d-shader: Fix saturates of fp64 types

### PR DESCRIPTION
**Pull request by Joshua-Ashton**
Closes: #419

Signed-off-by: Joshua Ashton <joshua@froggi.es>

### CODE REVIEW

File: libs/vkd3d-shader/spirv.c [hunk 1]
```diff
@@ -3716,14 +3716,24 @@ static uint32_t vkd3d_dxbc_compiler_emit_sat(struct vkd3d_dxbc_compiler *compile
...
-    zero_id = vkd3d_dxbc_compiler_get_constant_float_vector(compiler, 0.0f, component_count);
-    one_id = vkd3d_dxbc_compiler_get_constant_float_vector(compiler, 1.0f, component_count);
+    uint32_t type_id;
 
     type_id = vkd3d_dxbc_compiler_get_type_id_for_reg(compiler, reg, write_mask);
-    if (reg->data_type == VKD3D_DATA_FLOAT || reg->data_type == VKD3D_DATA_DOUBLE)
+
+    if (component_type == VKD3D_TYPE_DOUBLE)
+    {
+        uint32_t zero_id, one_id;
```
Comment (doitsujin) [inline libs/vkd3d-shader/spirv.c:3725]: any reason why we redeclare this in both branches?

Comment (Joshua-Ashton) [reply libs/vkd3d-shader/spirv.c:3725]: Just figured it'd be cleaner that way given it's only used within that scope.

Comment (doitsujin) [reply libs/vkd3d-shader/spirv.c:3725]: just leave the original declaration in place tbh, we do this in *some* cases but usually only for constants and if there's a lot of stuff that really is only needed on one side.

Or just write it as `uint32_t zero_id = ...`.

Comment (Joshua-Ashton) [reply libs/vkd3d-shader/spirv.c]: Resolved with the latter.

**Pull request merged**